app/code/Magento/Downloadable/Observer/UpdateLinkPurchasedObserver.php

```diff
@@ -0,0 +1,86 @@
+<?php
+/**
```

Comment:
Please use strict_types declaration for new implementations (http://php.net/manual/en/functions.arguments.php#functions.arguments.type-declaration.strict).
Example: https://github.com/magento/magento2/blob/2.3-develop/app/code/Magento/BundleGraphQl/Model/BundleProductTypeResolver.php#L6

Reply:
declare(strict_types=1); declaration added.